dspace-server-webapp/src/test/java/org/dspace/app/rest/converter/ConverterServiceIT.java

```diff
@@ -57,6 +62,16 @@
     @Mock
     private Object mockEmbeddedResource;
 
+    @Autowired
+    private RequestService requestService;
+
+    @Before
+    public void setup() {
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setAttribute("dspace.context", new Context());
+        MockHttpServletResponse mockHttpServletResponse = new MockHttpServletResponse();
+        requestService.startRequest(mockHttpServletRequest, mockHttpServletResponse);
```

Comment:
I don't understand the new `setup()` method here, as it doesn't seem to correspond with the new changes to `ConverterService` (but maybe I'm misunderstanding?).  Could you add comments as to why we need to mock the ServletRequest? (It's just not obvious to me, since the ServletRequest doesn't seem to be used in the new changes to ConverterService)

Reply:
yes, I also need help to understand the needs here and probably it will be good to add a javadoc or comment in the code for other that will check that at a later time.

It seems to me that you need to have a separate context bound to the "thread" for this IT but I don't see where this context is released

Reply:
I agree with @tdonohue and @abollini that adding comments would be helping to understand what is going on.

Reply:
This looks to have unresolved feedback.  I think I need inline comments here to understand why these changes are needed... as noted they don't seem to correspond to the changes you've made to `ConverterService` itself.

Reply:
Raf has included details explaining the change in https://github.com/DSpace/DSpace/commit/34bf7dd76b09bd20226e8b4a2317bae472eaf8d7#diff-91f223912e7443237a64a07d2d098f4fR70